cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -95,13 +121,22 @@ func (tcp *TestCloudProvider) Pricing() (cloudprovider.PricingModel, errors.Auto
 
 // GetAvailableMachineTypes get all machine types that can be requested from the cloud provider.
 func (tcp *TestCloudProvider) GetAvailableMachineTypes() ([]string, error) {
-	return []string{}, nil
+	return tcp.machineTypes, nil
 }
 
 // NewNodeGroup builds a theoretical node group based on the node definition provided. The node group is not automatically
 // created on the cloud provider side. The node group is not returned by NodeGroups() until it is created.
 func (tcp *TestCloudProvider) NewNodeGroup(machineType string, labels map[string]string, extraResources map[string]resource.Quantity) (cloudprovider.NodeGroup, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	return &TestNodeGroup{
+		cloudProvider:   tcp,
+		id:              "autoprovisioned-" + machineType,
+		minSize:         0,
+		maxSize:         1000,
+		targetSize:      0,
+		exist:           true,
```

Comment:
Shouldn't "exist" field be set to false here, as this is a theoretical node group? Based on Exist() semantics below: "Allows to tell the theoretical node group from the real one."

Reply:
This is a real (not autoprovisioned group) so it should exist.

Reply:
Fixed.